### ISSUE TITLE
isScalarType for non-null types

### ISSUE THREAD

**Issue by tomas-c** [graphql-java-kickstart/graphql-java-tools#335]
TLDR:

[isScalarType ](https://github.com/graphql-java-kickstart/graphql-java-tools/blob/742e62b806de81b619742442a8281de5bbf4db63/src/main/kotlin/com/coxautodev/graphql/tools/MethodFieldResolver.kt#L134) should support NotNullType. 

Without it, `mapper.convertValue(value, typeReference)` is always called and sometimes fails with latest Jackson.

Background:

After upgrading from Spring Boot 2.1.8 to 2.2.2 I started getting an exception when using ApolloScalars.Upload scalar type. The exception said:
```
No serializer found for class java.io.FileDescriptor and no properties discovered to create BeanSerializer
```
Turns our new version of Spring came with a new version of Jackson which had removed  [conversion short-circuiting](https://github.com/FasterXML/jackson-databind/commit/e3f005bf09975a3de674b44338bc6ca40f8c1efb)

**Comment by rrva** [comment]
https://github.com/graphql-java-kickstart/graphql-java-tools/pull/342 fixes this
